### PR TITLE
feat(general): TF and ARM - Ensure that Databricks Workspaces enable…

### PR DESCRIPTION
**Pull request by MrTomerLevi**
Ensure that Azure Databricks Workspaces enables customer-managed key for root DBFS encryption

**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sast|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

Ensure that Azure Databricks Workspaces enables customer-managed key for root DBFS encryption.

Fixes # (issue)

## New policies 

CKV2_AZURE_48 

### Description

This policy ensure that the root DBFS (Databricks File System) associated with Databricks Workspace is encrypted using a customer-managed key rather than the default Azure key.

https://learn.microsoft.com/en-us/azure/databricks/security/keys/customer-managed-keys-dbfs/
https://github.com/Azure/azure-quickstart-templates/blob/master/quickstarts/microsoft.databricks/databricks-workspace-with-dbfs-root-customer-managed-keys/azuredeploy.json

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/arm/checks/resource/DatabricksWorkspaceDBFSRootEncryptedWithCustomerManagedKey.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+from __future__ import annotations
+
+from typing import Any
+
+from checkov.common.models.enums import CheckResult, CheckCategories
+from checkov.arm.base_resource_check import BaseResourceCheck
+from checkov.common.util.data_structures_utils import find_in_dict
+
+# https://learn.microsoft.com/en-us/azure/templates/microsoft.databricks/workspaces?pivots=deployment-language-arm-template#workspaceencryptionparameter-1
```
Comment (bo156) [inline checkov/arm/checks/resource/DatabricksWorkspaceDBFSRootEncryptedWithCustomerManagedKey.py:9]: Perhaps add it as a documentation of the class?

File: checkov/terraform/checks/graph_checks/azure/DatabricksWorkspaceDBFSRootEncryptedWithCustomerManagedKey.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+          attribute: customer_managed_key_enabled
+          value: true
+        - resource_types:
+            - azurerm_databricks_workspace
+          connected_resource_types:
+            - azurerm_databricks_workspace_root_dbfs_customer_managed_key
+          operator: exists
+          cond_type: connection
+
+# Root DBFS encryption is only valid if the Databricks Workspace sku is set to 'premium'.
```
Comment (bo156) [inline checkov/terraform/checks/graph_checks/azure/DatabricksWorkspaceDBFSRootEncryptedWithCustomerManagedKey.yaml:44]: I think it's better if the comment is at the top of the file (unless it breaks something)

**Review by bo156 — APPROVED** [review]

**Review by tsmithv11 — APPROVED** [review]

File: checkov/terraform/checks/graph_checks/azure/DatabricksWorkspaceDBFSRootEncryptedWithCustomerManagedKey.yaml (hunk 2 above)
Comment (MrTomerLevi) [reply checkov/terraform/checks/graph_checks/azure/DatabricksWorkspaceDBFSRootEncryptedWithCustomerManagedKey.yaml:44]: I'm with you on this.
However, as I can see in other policies (i.e `AzureConfigMSSQLwithAD.yaml`, `AzureMariaDBserverUsingTLS_1_2.yaml`) the comment is placed in the bottom.

Let me know if you still want me to move it to the top

**Review by SteveVaknin — APPROVED** [review]
Great first contribution 🏆

**Pull request merged**